recipes/libcurl/all/conanfile.py

```diff
@@ -339,33 +339,53 @@ def _patch_cmake(self):
         # Use upstream FindZstd.cmake because check_symbol_exists() is called
         # afterwards and it would fail with zstd_LIBRARIES generated by CMakeDeps
         replace_in_file(self, cmakelists, "find_package(Zstd REQUIRED)", "find_package(Zstd REQUIRED MODULE)")
-        replace_in_file(self, os.path.join(self.source_folder, "CMake", "FindZstd.cmake"), "if(UNIX)", "if(0)")
+        if Version(self.version) < "8.10.0":
+            replace_in_file(self, os.path.join(self.source_folder, "CMake", "FindZstd.cmake"), "if(UNIX)", "if(0)")
 
         # c-ares
-        replace_in_file(self, cmakelists, "find_package(CARES REQUIRED)", "find_package(c-ares REQUIRED CONFIG)")
-        replace_in_file(self, cmakelists, "${CARES_LIBRARY}", "c-ares::cares")
+        if Version(self.version) < "8.10.0":
+            replace_in_file(self, cmakelists, "find_package(CARES REQUIRED)", "find_package(c-ares REQUIRED CONFIG)")
```

Comment:
@ErniGH, @AbrilRBS 
did we consider doing the overrides in the generate() method to avoid this patching?
https://docs.conan.io/2/reference/tools/cmake/cmakedeps.html#overwrite-properties-from-the-consumer-side-using-cmakedeps-set-property 

note that we may be able to also set `CARES_LIBRARY` via `tc.cache_variables` instead of patching the anything

I suspect the same could be done for a lot of the other dependencies

Alternative consider following the pattern of injecting a cmake file during `project()` so that we locate and define the relevant variables ahead of time? we could clean this up by A LOT :D

not critical...